hooks/tk-multi-publish2/basic/publish_session.py

```diff
@@ -0,0 +1,712 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import pprint
+import maya.cmds as cmds
+import maya.mel as mel
+import sgtk
+from sgtk.util.filesystem import copy_file
+import traceback
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class MayaSessionPublishPlugin(HookBaseClass):
+    """
+    Plugin for publishing an open maya session.
+    """
+
+    @property
+    def icon(self):
+        """
+        Path to an png icon on disk
+        """
+
+        # look for icon one level up from this hook's folder in "icons" folder
+        return os.path.join(
+            self.disk_location,
```

Comment:
Had to move some icons to the engine for this to work. Currently they live within the `hooks/tk-multi-publish2` folder, but perhaps we should standardize a level above like `hooks/icons` or maybe just use `resources/` at the engine root?

Reply:
Makes sense to me as engine's resources, but for when we move the default publishing logic into the engine.

Reply:
+1 - the `engine/resources` folder is intended for this kind of stuff!

Reply:
cool. will do.